pkg/client/typed/discovery/discovery_client.go

```diff
@@ -67,6 +67,8 @@ type ServerResourcesInterface interface {
 	ServerResourcesForGroupVersion(groupVersion string) (*unversioned.APIResourceList, error)
 	// ServerResources returns the supported resources for all groups and versions.
 	ServerResources() (map[string]*unversioned.APIResourceList, error)
+	// ServerNamespacedResources returns the supported namespaced resources for all groups and versions.
+	ServerNamespacedResources() ([]unversioned.GroupVersionResource, error)
```

Comment:
I wasn't sure, so I went with this approach first. Either way the name has to be aligned with the new functionality as well. I'm fine with ditching the old one entirely, in that case.